js/script.js

```diff
@@ -0,0 +1,56 @@
+window.onload = () => {
+    getData('home')
+}
+const sendReq = (METHOD, URL) => {
+    const promise = new Promise((resolve, reject) => {
+        const xhr = new XMLHttpRequest()
+        xhr.open(METHOD, URL)
+        xhr.send()
+        xhr.onload = () => {
+            resolve(xhr.response);
+        }
+    })
+    return promise
+}
+
+const getData = function (link) {
+    sendReq('GET', link + '.html').then(data => {
+        window.history.pushState({}, "", link)
+        if (link === 'home')
```

Comment:
what happens is the case your promise will reject?

Reply:
It will show message "Cannot get /{requestedURL}"
UPDATED: Now I have added catch block. This will show the user that request is invalid.